x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/onboarding/hooks/use_onboarding_success_callout.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { LOCAL_STORAGE_ONBOARDING_SUCCESS_CALLOUT_KEY } from '../../../constants';
+
+/**
+ * Hook to manage the visibility of the onboarding success callout
+ * initializing as false by default, so the callout is only displayed when manually enabled
+ * during the onboarding process.
+ */
+export const useOnboardingSuccessCallout = () => {
+  const [isCalloutVisible, setIsCalloutVisible] = useLocalStorage<boolean>(
+    LOCAL_STORAGE_ONBOARDING_SUCCESS_CALLOUT_KEY,
+    false
+  );
+
+  const onHideCallout = () => setIsCalloutVisible(false);
+
+  const dispatchSuccessCalloutVisibility = () => setIsCalloutVisible(true);
+
+  return {
+    isCalloutVisible,
+    onHideCallout,
+    dispatchSuccessCalloutVisibility,
+  };
```

Comment:
Nit but i think the namings for the two methods are a bit confusing and unnecessary since they are both only being used once, we can just expose `setIsCalloutVisible` directly.

For example, instead of
```
onDismiss={onHideCallout}
```

we can do
```
onDismiss={() => setIsCalloutVisible(false)}
```

If you still prefer to have the wrappers around the set functions, i would recommend to rename
```
onHideCallout -> hideCallout
dispatchSuccessCalloutVisibility -> showCallout
```

We can also just remove this hook completely since it doesn't add much, mainly a wrapper around the local storage key which is already a shared constant so its safe to reuse. 

either way this is just a nit pick since this doesn't have any broad effect so feel free to ignore

Reply:
Thanks, I prefer to keep the hook and the wrappers so the consumers and test files don't need to worry with the implementation details, but I updated the naming to be consistent